rfc/update-api.md

```diff
@@ -0,0 +1,105 @@
+# Update API
+
+The update API should expose easy, straightforward, extensible and forward compatible version information to the Constellation CLI (and possibly more consumers).
+
+Design goals:
+
+- Simple
+- Can be implemented using static HTTP file server
+- Generic over different kinds of resources
+    - OS image versions
+    - Microservice versions
+    - CLI versions
+    - Kubernetes versions
+
+The folowing HTTP endpoints are available:
+
+- `GET /constellation/v1/updates/latest/` contains files showing the latest version available
+    - `image.json` contains the latest image version
+    - `microservice.json` contains the latest microservice version
+    - `cli.json` contains the latest cli version
+    - `kubernetes.json` contains the latest supported version of Kubernetes
+- `GET /constellation/v1/updates/major/<major-version>/` contains files with version information for this major version
```

Comment:
I'm thinking about the tradeoff between defining different endpoints and just having one endpoint and do the version parsing client side. I think that one endpoint with all version is more generic, which benefits third parties. 

Also defining one endpoint and doing client side parsing allows for more flexibility.

Reply:
> Also defining one endpoint and doing client side parsing allows for more flexibility.

Multiple endpoints give us more flexibility for each resource type.
We could add metadata like changelogs, download URLs or other info that is unique to one kind of resource.